api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +539,349 @@ paths:
+              items:
+                description: list of subscriptions
+                type: array
+                items:
+                  $ref: '#/definitions/Subscription'
+            required:
+              - items
+              - _links
+
+  /subscriptions/{subscription_id}:
+    get:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Returns a subscription identified by id.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Subscription'
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions/{subscription_id}/cursors:
+    get:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Exposes the currently committed offsets of a subscription.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+      responses:
+        '200':
+          description: Ok
+          schema:
+            type: object
+            properties:
+              items:
+                description: list of cursors for subscription
+                type: array
+                items:
+                  $ref: '#/definitions/Cursor'
+            required:
+              - items
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+    post:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Endpoint for committing offsets of the subscription. The client must commit at least once
+        every 60 seconds, otherwise Nakadi will consider the client to be gone and will close the
+        connection.
+
+        When a batch is committed that also automatically commits all previous batches that were
+        sent in a stream for this partition.
+      parameters:
+        - name: subscription_id
+          in: path
+          type: string
+          description: Id of subscription
+          required: true
+        - name: X-Nakadi-StreamId
+          in: header
+          type: string
+          description: |
+            Id of stream which client uses to read events. It is not possible to make a commit for a terminated or
+            none-existing stream. Also the client can't commit something which was not sent to his stream.
+        - name: cursors
+          in: body
+          schema:
+            type: object
+            properties:
+              cursors:
+                description: |
+                  List of cursors that the consumer acknowledges to have successfully processed.
+                type: array
+                items:
+                  $ref: '#/definitions/Cursor'
+            required:
```

Comment:
Mixed topic streams are not supportable without adding something like an event type name to cursor sent back to the server (the API allows mixed topic subscriptions as a future extension). This will be a breaking change because subscription clients will be asked to send a new required field. I'd just make the addition now; btw sending the field now doesn't affect existing low level implementations using cursors, it's optional data to them.